### ISSUE TITLE
Broken tooltips

### ISSUE THREAD

**Issue by Drevil-Dion** [Lethay/TalentedClassic#96]
When talented for wrath classic and Zygor guides are both enabled tooltips for items players and everything are constnatly flashing upon disabling talented tooltips return to normal.

Talented disabled 
![image](https://user-images.githubusercontent.com/13806368/213525047-c6328216-be19-44c3-914b-83938cf672ad.png)

Talented enabled
![image](https://user-images.githubusercontent.com/13806368/213525411-681109d9-4d92-450b-91c4-308eae7b5297.png)

[Here is a gif showing the tooltip issue ](https://i.imgur.com/hzEes5s.gif)

No errors are shown at all had to manually go through addons one by one to find the cause which is only happening when Zygor guides and Talented are enabled.

Here is a screenshot of the talented.toc files showing which version i have which is v221207-WotLK uploaded to curse on Dec 7, 2022
![image](https://user-images.githubusercontent.com/13806368/213526923-bc71d470-b5e2-425b-abc6-0e802c97d19b.png)

**Comment by wazerstar** [comment]
I can confirm it breaks tooltips.

**Comment by Drevil-Dion** [comment]
> I can confirm it breaks tooltips.
> 
> Specially if you have zygor installed too.

Glad to know its not just me then Seems to have only happened with the latest update to p2 of wrath where they have introduced code from retail client to enable things like Heroic+

**Comment by Lethay** [comment]
Strange. I don't think I'm touching tooltips. Do many people use Zygor's addon?

**Comment by Drevil-Dion** [comment]
> Strange. I don't think I'm touching tooltips. Do many people use Zygor's addon?

Lot of players use zygor along with other similar add-ons like restedxp, rested however do not cause issues took me over a hour to figure out it was talented causing the issue due to the number of add-ons I have.

**Comment by Drevil-Dion** [comment]
Shows how big of a player base uses zygor avg salary for people working for them is 47k USD a year per person 7+ people employed.

https://i.imgur.com/ndNzt2O.png

**Comment by wazerstar** [comment]
> Strange. I don't think I'm touching tooltips. Do many people use Zygor's addon?

Yes Zygor is big, paid addon for guides, its been around since the the old Retail late TBC where carbonite got left in the dust.

**Comment by Drevil-Dion** [comment]
> > Strange. I don't think I'm touching tooltips. Do many people use Zygor's addon?
> 
> Yes Zygor is big, paid addon for guides, its been around since the the old Retail late TBC where carbonite got left in the dust.
> 
> Let me know if you need a test version to test with.

Free version can be downloaded from zygors site to test with 
https://zygorguides.com/join

**Comment by Lethay** [comment]
This'll probably take extensive testing, since I don't actually do anything to tooltips in my addon. I'm unlikely to get to this within the next year or so, since I have problems intrinsic to the addon to deal with first and I've only played the game for a week since WotLK released. But hopefully I'll get to it eventually.

**Comment by pquerner** [comment]
I've deleted all of the folders in /Libs and it works again. So yea, probably outdated Libs?

**Comment by Zaeyne** [comment]
It seems you don't need to delete the Talented>Libs folder completely, preventing LibGratuity-3.0 from being loaded worked for me.

Open Talented.toc and comment out this line:

Libs\LibGratuity-3.0\LibGratuity-3.0.lua

From what I can tell it doesn't affect the functionality of the addon.

**Comment by Drevil-Dion** [comment]
> It seems you don't need to delete the Talented>Libs folder completely, preventing LibGratuity-3.0 from being loaded worked for me.
> 
> Open Talented.toc and comment out this line:
> 
> Libs\LibGratuity-3.0\LibGratuity-3.0.lua
> 
> From what I can tell it doesn't affect the functionality of the addon.

Can confirm this works  talented loaded and zygor and tooltips showing perfect

![image](https://user-images.githubusercontent.com/13806368/213884332-b9b9f70e-0341-4741-9ea6-325c885bba8c.png)

**Comment by Iowerth** [comment]
Can confirm about tooltip issue and about fix. Thanks guys!

**Comment by Drevil-Dion** [comment]
> Any ETA on a release with this fix built in?

>  I'm unlikely to get to this within the next year or so, since I have problems intrinsic to the addon to deal with first and I've only played the game for a week since WotLK released. But hopefully I'll get to it eventually.

Takes a few moments to edit out the lines manually.

**Comment by Lethay** [comment]
Hello, I've pushed an update. Could you let me know if it worked?

**Comment by Lethay** [comment]
Ack. Doesn't look hard to fix but I'll need to be able to boot the game to debug it. Bear with me. Life's a bit busy, hence why I tried to go for the "easy" route of pushing without a test.

**Comment by Drevil-Dion** [comment]
> Ack. Doesn't look hard to fix but I'll need to be able to boot the game to debug it. Bear with me. Life's a bit busy, hence why I tried to go for the "easy" route of pushing without a test.
> 
> Incidentally, the only change I made since the last time I played the game myself was to delete LibGratuity. I simply had some uncommitted updates (namely, updating all the Ace libraries). So, it's a bit mystifying that it didn't work.

I would just push a update with those line myself and others have manually deleted for the time being since deleting them I have not had any issues at all

**Comment by tstirrat** [comment]
It works perfectly fine with the LibGratuity lib removed from `Talented.toc`, can you push an update with this? The lib does not seem to be used anywhere in talented (except commented out code).

**Comment by Werwolfsoul** [comment]
can confirm the issue repro and the manual fix by commenting out the line worked on version from 12/7/2022

**Comment by Lethay** [comment]
I've been away for a while and just got to testing this ingame myself. It worked perfectly. I even went and signed up to Zygor and installed their application, and it still worked. 

Then I went and had a look to see if the github files were somehow different.
...Yes, for quite some time, I'd had some files accidentally on gitgnore. This'll probably fix lot of things that've been broken for a while. Could someone test it again, to make sure it's not just me?

**Comment by wazerstar** [comment]
> I've been away for a while and just got to testing this ingame myself. It worked perfectly. I even went and signed up to Zygor and installed their application, and it still worked.
> 
> Then I went and had a look to see if the github files were somehow different. ...Yes, for quite some time, I'd had some files accidentally on gitgnore. This'll probably fix lot of things that've been broken for a while. Could someone test it again, to make sure it's not just me?

I just did a new test and this version you put it does not have any issues so far.

**Comment by Lethay** [comment]
Fabulous, I'll make it a release tomorrow. Thank you.

**Comment by AngelosNaoumis** [comment]
This is still present :(

**Comment by Lethay** [comment]
On what version of the game and with which version of the addon? Did it work with a previous version?

**Comment by AngelosNaoumis** [comment]
I am a new user so not very sure.

Currently using latest version of Zygor Version: 1.1.29966 and v231130-Era

Let me try version v230930-Era

**Comment by AngelosNaoumis** [comment]
Still happening with v230930-Era aswell

**Comment by Lethay** [comment]
Aha. It was a problem with the WotLK version which we then fixed. Apparently it's here with the Era version too. It's related to LibGratuity but unfortunately LibGratuity is required for this version of the addon without a rewrite. The rewrite required functions that WotLK has but Vanilla doesn't. I'll have to see if it's possible to use something other than LibGratuity.
